src/toolbars/EditToolBar.cpp

```diff
@@ -129,49 +173,49 @@ void EditToolBar::Populate()
    /* Buttons */
    // Tooltips match menu entries.
    // We previously had longer tooltips which were not more clear.
-   AddButton(this, bmpZoomIn, bmpZoomIn, bmpZoomInDisabled, ETBZoomInID,
+   AddButton(bmpZoomIn, bmpZoomIn, bmpZoomInDisabled, ETBZoomInID,
```

Comment:
I like the effort to unify repepetious code into a utility class.

It might go farther (in 3.3 development).

The `Populate()` routines might also become completely table-driven if `Entry` could indicate a separator and also what the initial enabled state should be.

The `_EVENT_TABLE` macros might be removed and instead `ToolBarButtons` could `Bind` an event handler to each button as it is made.  (The preferred thing in newer wxWidgets code)

It might even be not too difficult to devise a platform-neutral facade class to unify these two simple toolbars, and also use a registry of button entries so that an add-on might insert buttons.